src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -4797,7 +4798,7 @@ void ClassFileParser::verify_legal_method_modifiers(jint flags,
           // ACC_STRICT, or ACC_SYNCHRONIZED flags set.  No need to
           // check for ACC_FINAL, ACC_NATIVE or ACC_SYNCHRONIZED as
           // those flags are illegal irrespective of ACC_ABSTRACT being set or not.
-          (is_abstract && (is_private || is_static || is_strict))) {
+          (is_abstract && (is_private || is_static || (!major_gte_17 && is_strict)))) {
```

Comment:
I'm not sure it makes sense to allow ACC_STRICT to suddenly appear where it would otherwise be illegal. ??